src/main/java/it/units/sdm/project/interfaces/Board.java

```diff
@@ -0,0 +1,19 @@
+package it.units.sdm.project.interfaces;
+
+import it.units.sdm.project.Position;
+import org.jetbrains.annotations.NotNull;
+
+
+public interface Board {
```

Comment:
You said: *I thought that in general, a Board may have more specific methods depending on the used pieces. For example, in our case, we're using black and white Stones, therefore we have the putStone() and getStone() methods*
I think that the Board and Cell classes are the same for all the games. Instead, we can generalize Stone, so we will have putSomething and getSomething. I would prefer to build an util class, instead, to implement more specific game related methods. The Board must not be dependent on the type of the games.

Reply:
I guess we could do something similar to the `List<T>` java interface, where `T` would be `Stone` in our case

Reply:
Yes, great idea!

Reply:
I uploaded a first attempt, but as you can see and as I feared there's not much improvement regarding readability and overall class structure. The `FreedomCell` class and `Cell` interface are now basically the same aside from the `stone` status variable

Reply:
I see, but as I told you before I don't think that Board has to be an interface (neither Cell). If we are going to implement other games, we have to ovveride the same methods that would have almost the same implementation.
We will talk about it tonight on the meeting I think.

Reply:
> Yes, great idea!

Actually not, lol (at least for me).